### PR TITLE
feat(sdk/go): implement sqlite SDK for TinyGo

### PR DESCRIPTION
**Pull request by adamreese**
Implements Golang sql database interface for Spin's sqlite.

### CODE REVIEW

File: sdk/go/sqlite/sqlite.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+// QueryContext executes a query that may return rows, such as a SELECT.
+func (c *Conn) QueryContext(_ context.Context, query string, args []driver.Value) (driver.Rows, error) {
+	return c.Query(query, args)
+}
+
+// Exec isn't implemented. Use Query method.
+func (c *Conn) Exec(_ context.Context, _ string, _ []driver.NamedValue) (driver.Result, error) {
+	return nil, errors.New("Exec method not implemented")
+}
```
Comment (rylev) [inline sdk/go/sqlite/sqlite.go:46]: I there a reason we're not implementing this?

Comment (adamreese) [reply sdk/go/sqlite/sqlite.go:46]: Correct me if I'm wrong... the returned object should be `LastInsertId` and `RowsAffected` which I don't think is returned from our implementation. https://pkg.go.dev/database/sql/driver#Result

**Comment by adamreese** [comment]
Pushed up some changes to support `Prepare()`

File: sdk/go/sqlite/sqlite.go (hunk 1 above)
Comment (rylev) [reply sdk/go/sqlite/sqlite.go]: Ah you're correct. Sorry for the noise!

**Review by rylev — APPROVED** [review]
This looks good to me - I think we should get this in and have others write some code against it.

File: sdk/go/sqlite/sqlite.go (hunk 1 above)
Comment (adamreese) [reply sdk/go/sqlite/sqlite.go]: I missed the note in the doc that says `Not every database or database driver may support this.` So I'll add it in

**Comment by adamreese** [comment]
@rylev I added support for the `Exec` methods and removed some unneeded Context methods.  Can I get a fresh LGTM?

**Pull request merged**